lib/rbvmomi/basic_types.rb

```diff
@@ -201,6 +199,36 @@ def _ref
         @ref
       end
 
+      def to_s
+        "#{self.class.wsdl_name}(#{@ref.inspect})"
+      end
+
+      def to_hash
+        to_s
+      end
+
+      def pretty_print pp
+        pp.text to_s
+      end
+
+      def == x
+        out = (x.class == self.class && x._ref == @ref)
```

Comment:
Not for this PR, but It think this can be simplified to

```suggestion
        out = (x.hash == hash)
```

On that note, it's curious that .hash doesn't consider `instanceUuid`, but `==` does

Reply:
This change would also make it so you wouldn't need to expose the _ref, which seems like it's only there as like a "friend-private" methods (I'm guessing because of the `_ ` prefix).

Reply:
Actually `_ref` is used quite often by callers, the `_` prefix is so that it won't conflict with any wsdl property named `ref` since we dynamically define accessors for wsdl properties

Reply:
> On that note, it's curious that .hash doesn't consider instanceUuid, but == does

If I had to guess it might be because `_connection.instanceUuid` could result in an API call to retrieveServiceContent